java-runtime/ftl-runtime/deployment/src/main/java/xyz/block/ftl/deployment/VerbClientsProcessor.java

```diff
@@ -0,0 +1,159 @@
+            } else {
+                classOutput = new GeneratedClassGizmoAdaptor(generatedClients, true);
+            }
+            //TODO: map and list return types
+            for (var i : iface.interfaceTypes()) {
+                if (i.name().equals(VERB_CLIENT)) {
+                    if (i.kind() == Type.Kind.PARAMETERIZED_TYPE) {
+                        var returnType = i.asParameterizedType().arguments().get(1);
+                        var paramType = i.asParameterizedType().arguments().get(0);
+                        try (ClassCreator cc = new ClassCreator(classOutput, iface.name().toString() + "_fit_verbclient", null, Object.class.getName(), iface.name().toString())) {
+                            if (launchModeBuildItem.isTest()) {
+                                cc.addAnnotation(TEST_ANNOTATION);
+                                cc.addAnnotation(Singleton.class);
+                            }
+                            var publish = cc.getMethodCreator("call", returnType.name().toString(), paramType.name().toString());
+                            var helper = publish.invokeStaticMethod(MethodDescriptor.ofMethod(VerbClientHelper.class, "instance", VerbClientHelper.class));
+                            var results = publish.invokeVirtualMethod(MethodDescriptor.ofMethod(VerbClientHelper.class, "call", Object.class, String.class, String.class, Object.class, Class.class, boolean.class, boolean.class), helper, publish.load(name), publish.load(module), publish.getMethodParam(0), publish.loadClass(returnType.name().toString()), publish.load(false), publish.load(false));
+                            publish.returnValue(results);
+                            publish = cc.getMethodCreator("call", Object.class, Object.class);
+                            helper = publish.invokeStaticMethod(MethodDescriptor.ofMethod(VerbClientHelper.class, "instance", VerbClientHelper.class));
+                            results = publish.invokeVirtualMethod(MethodDescriptor.ofMethod(VerbClientHelper.class, "call", Object.class, String.class, String.class, Object.class, Class.class, boolean.class, boolean.class), helper, publish.load(name), publish.load(module), publish.getMethodParam(0), publish.loadClass(returnType.name().toString()), publish.load(false), publish.load(false));
+                            publish.returnValue(results);
+                            clients.put(iface.name(), new VerbClientBuildItem.DiscoveredClients(name, module,cc.getClassName()));
+                        }
+                        found = true;
+                        break;
+                    } else {
+                        throw new RuntimeException("@VerbClientDefinition can only be applied to interfaces that directly extend a verb client type with concrete type parameters and " + iface.name() + " does not have concrete type parameters");
+                    }
```

Comment:
Could all these if blocks be refactored out into a factory of sorts? There's a lot of very similar calls and variables. Maybe some DRY if it's appropriate?

Reply:
Even though it all looks similar each line is different enough that it can't really be turned generic. Basically this is due to the fact that there is 4 different client signatures, and each once needs a slightly different implementation.